### PR TITLE
fixes #4111 - Add image user data flag to API

### PR DESCRIPTION
**Pull request by shlomizadok**
fixes #4111

### CODE REVIEW

**Comment by ohadlevy** [comment]
[test]

**Comment by isratrade** [comment]
@shlomizadok, looks good. Can you squash the two commit into one and have only one line break at end of file.

File: app/views/api/v2/images/main.json.rabl [hunk 1]
```diff
@@ -5,5 +5,10 @@ extends "api/v2/images/base"
 attributes :operatingsystem_id, :operatingsystem_name, :compute_resource_id, :compute_resource_name,
            :architecture_id, :architecture_name, :uuid, :username, :created_at, :updated_at
 
+attribute :user_data, :if => lambda { |img| (img.compute_resource.kind_of? Foreman::Model::EC2) ||
```
Comment (domcleal) [inline app/views/api/v2/images/main.json.rabl]: This isn't very flexible, it should use fragments per compute resource like app/views/api/v2/compute_resources/ so plugins can extend it.

Comment (shlomizadok) [inline app/views/api/v2/images/main.json.rabl]: @domcleal -- good point. refactoring.

**Comment by shlomizadok** [comment]
Changed so is can be more flexible. I have added an if that checks whether the provider is EC2, Rackspace, OpenStack so we won't have to add a (blank) file per each provider. cc @domcleal

File: app/views/api/v2/images/main.json.rabl [hunk 2]
```diff
@@ -5,5 +5,14 @@ extends "api/v2/images/base"
 attributes :operatingsystem_id, :operatingsystem_name, :compute_resource_id, :compute_resource_name,
            :architecture_id, :architecture_name, :uuid, :username, :created_at, :updated_at
 
+
+
+if @image && ((@image.compute_resource.kind_of? Foreman::Model::EC2) || (@image.compute_resource.kind_of? Foreman::Model::Rackspace) || (@image.compute_resource.kind_of? Foreman::Model::Openstack))
```
Comment (ohadlevy) [inline app/views/api/v2/images/main.json.rabl]: this is not portable, e.g. a plugin cant extend / use it.

another option could be to add a method on the compute resource, similar to capability ?

**Comment by isratrade** [comment]
@shlomizadok, looks good.  Squash the 3 commits into 1 and then it looks ready for merge.

File: test/unit/compute_resource_test.rb [hunk 3]
```diff
@@ -102,4 +102,9 @@ def setup
     assert_equal [taxonomies(:location1).id], compute_resources(:one).used_or_selected_location_ids
   end
 
+  test "user_data_supported?" do
+    assert_equal false, compute_resources(:one).user_data_supported?
```
Comment (dLobatog) [inline test/unit/compute_resource_test.rb]: Please use `refute` and `assert` instead of `assert_equal boolean`

Comment (shlomizadok) [inline test/unit/compute_resource_test.rb]: :+1:

**Comment by dLobatog** [comment]
Reviewed and tested, to me it's nearly ready to merge, @shlomizadok, please remove doc/apipie_examples.json from the commit.

**Comment by isratrade** [comment]
merged as d21103bcf13b5981601be88330ce73dbe4a1ed77. Thanks @shlomizadok !

**Pull request closed without merging**